scripts/prepare_release.rb

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env ruby
+
+require 'octokit'
+require 'fileutils'
+require 'erb'
+require 'colorize'
+require 'optparse'
+
+def prompt_for_new_version
+  puts "What's the new version number?"
+  version = gets.chomp
+end
+
+def validate_version_number(version_number)
+  part_names = ['major', 'minor', 'patch']
+  parts = version_number.split('.')
+
+  unless parts.length() == 3
+    abort("Invalid version number. It should consists of a major, minor, and patch number.")
+  end
+
+  parts.each_with_index do | part, index |
+    if part.start_with?('0') && part.length() > 1
+      part_name = part_names[index]
+      abort("Invalid version number: #{part_name} number can\'t begin with 0.")
+    end
+  end
+end
+
+def update_changelog(version)
+  done = false
+  final_lines = []
+
+  File.foreach('CHANGELOG.md') { |line|
+    is_line = line.include? "XX.XX.XX"
+    final_lines.append(line)
+
+    date = Time.now.strftime("%Y-%m-%d")
+
+    if is_line
+        final_lines.append("\n")
+        final_lines.append("## #{version} - #{date}\n")
+    end
+  }
+
+  File.write('CHANGELOG.md', final_lines.join(""))
+end
+
+def open_url(url)
+  `open '#{url}'`
+end
+
+# Script start
+
+new_version_number = prompt_for_new_version
+validate_version_number(new_version_number)
+
+new_branch_name = "release/#{new_version_number}"
+action_title = "Update release notes for #{new_version_number}"
+
+# Pull the current state
+system("git checkout master")
```

Comment:
Should we ensure git is clean at the start of the script?

Reply:
I added a modified version of [this check](https://git.corp.stripe.com/stripe-internal/bindings/blob/master/lib/bindings.rb#L137).